integration/tests/observer/observer_test.go

```diff
@@ -0,0 +1,274 @@
+package observer
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/rs/zerolog"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/suite"
+	"google.golang.org/grpc"
+
+	"github.com/onflow/flow-go/integration/testnet"
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/utils/unittest"
+	accessproto "github.com/onflow/flow/protobuf/go/flow/access"
+)
+
+func TestObserver(t *testing.T) {
+	suite.Run(t, new(ObserverSuite))
+}
+
+type ObserverSuite struct {
+	suite.Suite
+	net      *testnet.FlowNetwork
+	teardown func()
+}
+
+func (suite *ObserverSuite) TearDownTest() {
+	suite.teardown()
+}
+
+func (suite *ObserverSuite) SetupTest() {
+	nodeConfigs := []testnet.NodeConfig{
+		// access node with unstaked nodes supported
+		testnet.NewNodeConfig(flow.RoleAccess, testnet.WithLogLevel(zerolog.InfoLevel), func(nc *testnet.NodeConfig) {
+			nc.SupportsUnstakedNodes = true
+		}),
+		// need one dummy execution node (unused ghost)
+		testnet.NewNodeConfig(flow.RoleExecution, testnet.WithLogLevel(zerolog.FatalLevel), testnet.AsGhost()),
+		// need one dummy verification node (unused ghost)
+		testnet.NewNodeConfig(flow.RoleVerification, testnet.WithLogLevel(zerolog.FatalLevel), testnet.AsGhost()),
+		// need one controllable collection node (unused ghost)
+		testnet.NewNodeConfig(flow.RoleCollection, testnet.WithLogLevel(zerolog.FatalLevel), testnet.AsGhost()),
+	}
+
+	// need three consensus nodes (unused ghost)
+	for n := 0; n < 3; n++ {
+		conID := unittest.IdentifierFixture()
+		nodeConfig := testnet.NewNodeConfig(flow.RoleConsensus,
+			testnet.WithLogLevel(zerolog.FatalLevel),
+			testnet.WithID(conID),
+			testnet.AsGhost())
+		nodeConfigs = append(nodeConfigs, nodeConfig)
+	}
+
+	// prepare the network
+	conf := testnet.NewNetworkConfig("observer_api_test", nodeConfigs)
+	suite.net = testnet.PrepareFlowNetwork(suite.T(), conf, flow.Localnet)
+
+	// start the network
+	ctx := context.Background()
+	suite.net.Start(ctx)
+
+	stop, err := suite.net.AddObserver(ctx, &testnet.ObserverConfig{
+		ObserverName:            "observer_1",
+		ObserverImage:           "gcr.io/flow-container-registry/observer:latest",
+		AccessName:              "access_1",
+		AccessPublicNetworkPort: fmt.Sprint(testnet.AccessNodePublicNetworkPort),
+		AccessGRPCSecurePort:    fmt.Sprint(testnet.DefaultSecureGRPCPort),
+	})
+
+	if err != nil {
+		// this can happen occaisionally...
+		// usually it's because docker didn't remove the observer container during the previous test.
+		// the observer container is removed by an "AutoRemove" flag in AddObserver.
+		panic(err)
+	}
+
+	time.Sleep(time.Second * 3) // needs breathing room for the observer to start listening
```

Comment:
This is set to 5 seconds in execution_state_sync_test.go, why is it 3 here for what looks like the same operation? If setting up an observer is something we do often could this code be placed in a utility to prevent code desync?

Reply:
Thank you for the review @AndrewM-SDET. Fixed: [befe665](https://github.com/GetElastech/flow-go/commit/befe6658e16bfe0e0c2e02677b9bd1904072f12f)

Reply:
I was basically exercising the tests and the experimental value remained there.